smac_planner/src/a_star.cpp

```diff
@@ -315,42 +318,24 @@ void AStarAlgorithm::getNeighbors(const unsigned int & node, NodeVector & neighb
   // 100 100 100   where lower-middle '100' is visited with same cost by both bottom '50' nodes
   // Therefore, it is valuable to have some low-potential across the entire map
   // rather than a small inflation around the obstacles
-  const int node_i = static_cast<int>(node);
 
   switch (_neighborhood) {
```

Comment:
I think the neighborhood type should become a property of the node. In the most extreme case, we could have 2 different Nodes that 1 is 4 and the other is 8 connected. Functionally, I think we can make just 1 Node that supports both of these based on the enum. I'm thinking about the case later when we truly do have multiple Node templates and we need the general `getNeighbors` function to support other types